File: runtime/CompressionInterceptor.java
package okhttp3;

import java.io.IOException;

public class CompressionInterceptor implements Interceptor {

    public enum DecompressionAlgorithm {
        BROTLI,
        GZIP,
        ZSTD
    }

    public CompressionInterceptor(DecompressionAlgorithm... algorithms) {}

    @Override
    public Response intercept(Chain chain) throws IOException {
        return chain.proceed(chain.request());
    }
}

File: android/app/src/main/kotlin/okhttp3/CompressionInterceptor.kt
package okhttp3

import okio.BufferedSource

class CompressionInterceptor(vararg algorithms: DecompressionAlgorithm) : Interceptor {
    interface DecompressionAlgorithm {
        fun name(): String
        fun decompress(source: BufferedSource): BufferedSource
    }

    override fun intercept(chain: Interceptor.Chain): Response {
        return chain.proceed(chain.request())
    }
}
